include/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesBroker.hpp

```diff
@@ -0,0 +1,82 @@
+/// Apache License 2.0
+
+#ifndef __OpenSpaceToolkit_Astrodynamics_Trajectory_CoordinatesBroker__
+#define __OpenSpaceToolkit_Astrodynamics_Trajectory_CoordinatesBroker__
+
+#include <OpenSpaceToolkit/Core/Types/Integer.hpp>
+#include <OpenSpaceToolkit/Core/Types/String.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesSubset.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace trajectory
+{
+
+using ostk::core::types::Integer;
+
+using ostk::astro::trajectory::CoordinatesSubset;
+
+/// @brief  State coordinates broker.
+
+class CoordinatesBroker
+{
+   public:
+    /// @brief              Constructor
+    ///
+    /// @code
+    ///                     CoordinatesBroker coordinatesSubset() ;
+    /// @endcode
+
+    CoordinatesBroker();
+
+    /// @brief              Adds a coordinates subset to be considered, returning the starting index it will occupy (or
+    /// that it occupies if it was already added) in the state coordinates
+    ///
+    /// @param              [in] aCoordinatesSubset a coordinate subset to be considered
+    ///
+    /// @return             The starting index of the subset in the state coordinates
+
+    Integer addSubset(const CoordinatesSubset& aCoordinatesSubset);
```

Comment:
Consider using the `Size` type from `core` for indices since it's unsigned
```suggestion
    Size addSubset(const CoordinatesSubset& aCoordinatesSubset);
```